src/base/net/downloadhandler.cpp

```diff
@@ -47,16 +47,26 @@ static QString errorCodeToString(QNetworkReply::NetworkError status);
 
 using namespace Net;
 
-DownloadHandler::DownloadHandler(QNetworkReply *reply, DownloadManager *manager, bool saveToFile, qint64 limit, bool handleRedirectToMagnet)
+DownloadHandler::DownloadHandler(QNetworkReply *reply, DownloadManager *manager, const DownloadRequest &downloadRequest)
     : QObject(manager)
     , m_reply(reply)
     , m_manager(manager)
-    , m_saveToFile(saveToFile)
-    , m_sizeLimit(limit)
-    , m_handleRedirectToMagnet(handleRedirectToMagnet)
-    , m_url(reply->url().toString())
+    , m_downloadRequest(downloadRequest)
+    , m_url(reply ? reply->url().toString() : downloadRequest.url)
 {
-    init();
+    if (reply)
+        assignNetworkReply(reply);
+}
+
+void DownloadHandler::assignNetworkReply(QNetworkReply *reply)
```

Comment:
This method is only an implementation detail. It is only used twice (and unlikely to be used anywhere else). I can even remove the assertion because I'm not too worried that it can be misused.
Of course I can change this to a reference.